pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseSingleBlockCombineOperator.java

```diff
@@ -111,13 +105,11 @@ protected void processSegments() {
           ((AcquireReleaseColumnsSegmentOperator) operator).release();
         }
       }
-
+      _blockingQueue.offer(resultsBlock);
+      // When query is satisfied, skip processing the remaining segments
       if (_resultsBlockMerger.isQuerySatisfied(resultsBlock)) {
-        // Query is satisfied, skip processing the remaining segments
-        _blockingQueue.offer(resultsBlock);
+        _nextOperatorId.set(_numOperators);
```

Comment:
hmm. i am a bit worry about this one. 
basically, this will make numBlocksMerged to practically impossible to reach numOperators during mergeResultBlock method. 
so we are solely relying on the early termination mechanism to also early terminate on the mergeResultBlock.

however the 
`_resultsBlockMerger.isQuerySatisfied(resultsBlock)` is not the same as the one in merge method
`_resultsBlockMerger.isQuerySatisfied(mregedBlock)`
are we 100% sure these are equivalent for sure?

Reply:
Yes. When any single block can satisfy the query, the merged block can always satisfy the query.
Before #9387 we actually pre-assign segments to the thread, so we never rely on all operators to be finished.

Reply:
sounds good. i think the ResultBlockerMerger javadoc also stated that.